### PR TITLE
chore(checkbox): update token variables

### PR DESCRIPTION
**Pull request by pixelflips**
# Description
Update checkbox tokens to use new core/semantic values where applicable.

Fixes [DSS-563](https://kajabi.atlassian.net/browse/DSS-563)

## Type of change
- [x] Token adjustments

# How Has This Been Tested?

Please describe the tests you've added and run to verify your changes.
Provide instructions so that we can reproduce.
Please also list any relevant details for your test configuration.

- [ ] unit tests
- [ ] e2e tests
- [ ] accessibility tests
- [x] tested manually
- [ ] other:

**Test Configuration**:

- Pine versions:
- OS:
- Browsers:
- Screen readers:
- Misc:

# Checklist:

If not applicable, leave options unchecked.

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [ ] Design has QA'ed and approved this PR

[DSS-563]: https://kajabi.atlassian.net/browse/DSS-563?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by ju-Skinner — APPROVED** [review]

File: libs/core/src/components/pds-checkbox/pds-checkbox.scss [hunk 1]
```diff
@@ -1,21 +1,21 @@
 :host {
-  --background-color: var(--pine-color-base-white);
-  --background-color-disabled: var(--pine-color-neutral-grey-200);
-  --background-color-hover: var(--pine-color-neutral-grey-100);
-  --border: var(--pine-border-interactive);
-  --border-color-disabled: var(--pine-color-neutral-charcoal-100);
-  --border-color-hover: var(--pine-border-interactive-hover-color);
-  --border-color-icon: var(--pine-color-base-white);
-  --border-radius: var(--pine-border-radius-xs);
+  --border: var(--pine-border-width-thin) solid var(--pine-color-grey-400);
```
Comment (anechol) [inline libs/core/src/components/pds-checkbox/pds-checkbox.scss]: Could we add `default` here at the end? What do you think?

File: libs/core/src/components/pds-checkbox/pds-checkbox.scss [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
+  --border-color-disabled: var(--pine-color-charcoal-100);
+  --border-color-icon: var(--pine-color-white);
+  --border-color-interactive-hover: var(--pine-border-width-thin) solid var(--pine-color-grey-500);
+  --border-radius: var(--pine-border-radius-050);
   --box-shadow-focus: 0 0 0 2px var(--pine-color-primary-200);
   --box-shadow-focus-invalid: 0 0 0 2px var(--pine-color-red-200);
-  --color-checked: var(--pine-color-neutral-charcoal-500);
-  --color-disabled: var(--pine-color-neutral-grey-300);
+  --color-background: var(--pine-color-white);
```
Comment (anechol) [inline libs/core/src/components/pds-checkbox/pds-checkbox.scss:7]: Here as well

File: libs/core/src/components/pds-checkbox/pds-checkbox.scss [hunk 3]
```diff
@@ -1,21 +1,21 @@
...
+  --color-background-disabled: var(--pine-color-grey-200);
+  --color-background-hover: var(--pine-color-grey-100);
+  --color-checked: var(--pine-color-charcoal-500);
+  --color-disabled: var(--pine-color-grey-300);
   --color-focus: var(--pine-color-primary-200);
-  --color-invalid-focus: var(--pine-color-red-200);
   --color-invalid: var(--pine-color-red-300);
-  --message-color-disabled: var(--pine-color-neutral-grey-500);
-  --message-color: var(--pine-color-neutral-charcoal-200);
+  --color-invalid-focus: var(--pine-color-red-200);
```
Comment (anechol) [inline libs/core/src/components/pds-checkbox/pds-checkbox.scss]: Can we add `text` in here? Would help give better context on where that color is being applied.

File: libs/core/src/components/pds-checkbox/pds-checkbox.scss [hunk 4]
```diff
@@ -1,21 +1,21 @@
...
-  --background-color-hover: var(--pine-color-neutral-grey-100);
-  --border: var(--pine-border-interactive);
-  --border-color-disabled: var(--pine-color-neutral-charcoal-100);
-  --border-color-hover: var(--pine-border-interactive-hover-color);
-  --border-color-icon: var(--pine-color-base-white);
-  --border-radius: var(--pine-border-radius-xs);
+  --border: var(--pine-border-width-thin) solid var(--pine-color-grey-400);
+  --border-color-disabled: var(--pine-color-charcoal-100);
+  --border-color-icon: var(--pine-color-white);
+  --border-color-interactive-hover: var(--pine-border-width-thin) solid var(--pine-color-grey-500);
```
Comment (anechol) [inline libs/core/src/components/pds-checkbox/pds-checkbox.scss]: These can be tricky, but do you think based on the application, that these could better fit under the color category?

**Review by anechol — CHANGES_REQUESTED** [review]
Nice work on this! Left some nitpicky comments on things I think could be addressed.

File: libs/core/src/components/pds-checkbox/pds-checkbox.scss (hunk 1 above)
Comment (pixelflips) [reply libs/core/src/components/pds-checkbox/pds-checkbox.scss]: Works for me!

File: libs/core/src/components/pds-checkbox/pds-checkbox.scss (hunk 2 above)
Comment (pixelflips) [reply libs/core/src/components/pds-checkbox/pds-checkbox.scss:7]: Not sure I understand the ask here.

File: libs/core/src/components/pds-checkbox/pds-checkbox.scss (hunk 4 above)
Comment (pixelflips) [reply libs/core/src/components/pds-checkbox/pds-checkbox.scss]: Done

File: libs/core/src/components/pds-checkbox/pds-checkbox.scss (hunk 3 above)
Comment (pixelflips) [reply libs/core/src/components/pds-checkbox/pds-checkbox.scss]: the `color-invalid` applies to more than just text, but let me know if I am misunderstanding.

File: libs/core/src/components/pds-checkbox/pds-checkbox.scss (hunk 2 above)
Comment (anechol) [reply libs/core/src/components/pds-checkbox/pds-checkbox.scss:7]: Whoops sorry looks like my other comment didn't go through. I was going to ask to add context to where/how the color would be used, but it looks like you've updated that. All good.

File: libs/core/src/components/pds-checkbox/pds-checkbox.scss (hunk 3 above)
Comment (anechol) [reply libs/core/src/components/pds-checkbox/pds-checkbox.scss]: @pixelflips Hmm yep, I see your point and angle here. Originally, I intended to be more explicit with how/where certain tokens will be used but it does bring up the point about cases where some may/could be reused. I think I may have done this in a few places as well. I'd like to see what the team thinks about it overall, but let's get this in for now.

File: libs/core/src/components/pds-checkbox/pds-checkbox.scss [hunk 5]
```diff
@@ -1,21 +1,20 @@
...
-  --color-checked: var(--pine-color-neutral-charcoal-500);
-  --color-disabled: var(--pine-color-neutral-grey-300);
+  --color-background: var(--pine-color-white);
+  --color-background-disabled: var(--pine-color-grey-200);
+  --color-background-hover: var(--pine-color-grey-100);
+  --color-border-disabled: var(--pine-color-charcoal-100);
+  --color-border-icon: var(--pine-color-white);
+  --color-checked: var(--pine-color-charcoal-500);
+  --color-disabled: var(--pine-color-grey-300);
   --color-focus: var(--pine-color-primary-200);
```
Comment (anechol) [inline libs/core/src/components/pds-checkbox/pds-checkbox.scss]: Could we add more context here for these?

**Review by anechol — APPROVED** [review]
One small comment, non-blocking. Thanks for getting these updated!

File: libs/core/src/components/pds-checkbox/pds-checkbox.scss (hunk 5 above)
Comment (pixelflips) [reply libs/core/src/components/pds-checkbox/pds-checkbox.scss]: @anechol so for the `--color-checked` and `--color-disabled` those apply both a background and a border color. let me know how i should handle those.

**Review by ju-Skinner — APPROVED** [review]

**Pull request merged**